http://bugs.mysql.com/bug.php?id=58350

=== modified file 'sql/sp_head.cc'
--- a/sql/sp_head.cc	2010-11-18 15:01:58 +0000
+++ b/sql/sp_head.cc	2010-12-08 16:47:21 +0000
@@ -3159,7 +3159,7 @@ sp_instr_stmt::exec_core(THD *thd, uint 
   MYSQL_QUERY_EXEC_START(thd->query(),
                          thd->thread_id,
                          (char *) (thd->db ? thd->db : ""),
-                         thd->security_ctx->priv_user,
+                         &thd->security_ctx->priv_user[0],
                          (char *)thd->security_ctx->host_or_ip,
                          3);
   int res= mysql_execute_command(thd);

=== modified file 'sql/sql_connect.cc'
--- mysql-5.5.8/sql/sql_connect.cc.orig 2010-12-03 11:58:26.000000000 -0600
+++ mysql-5.5.8/sql/sql_connect.cc      2011-02-07 16:36:27.819580199 -0600
@@ -733,7 +733,7 @@
     if (login_connection(thd))
       goto end_thread;
 
-    MYSQL_CONNECTION_START(thd->thread_id, thd->security_ctx->priv_user,
+    MYSQL_CONNECTION_START(thd->thread_id, &thd->security_ctx->priv_user[0],
                            (char *) thd->security_ctx->host_or_ip);
 
     prepare_new_connection_state(thd);

=== modified file 'sql/sql_cursor.cc'
--- a/sql/sql_cursor.cc	2010-11-12 12:56:21 +0000
+++ b/sql/sql_cursor.cc	2010-12-08 16:47:21 +0000
@@ -111,7 +111,7 @@ int mysql_open_cursor(THD *thd, select_r
   MYSQL_QUERY_EXEC_START(thd->query(),
                          thd->thread_id,
                          (char *) (thd->db ? thd->db : ""),
-                         thd->security_ctx->priv_user,
+                         &thd->security_ctx->priv_user[0],
                          (char *) thd->security_ctx->host_or_ip,
                          2);
   rc= mysql_execute_command(thd);

=== modified file 'sql/sql_parse.cc'
--- a/sql/sql_parse.cc	2010-12-14 14:34:23 +0000
+++ b/sql/sql_parse.cc	2010-12-08 16:47:21 +0000
@@ -877,7 +877,7 @@ bool dispatch_command(enum enum_server_c
   thd->profiling.start_new_query();
 #endif
   MYSQL_COMMAND_START(thd->thread_id, command,
-                      thd->security_ctx->priv_user,
+                      &thd->security_ctx->priv_user[0],
                       (char *) thd->security_ctx->host_or_ip);
   
   thd->command=command;
@@ -1018,7 +1018,7 @@ bool dispatch_command(enum enum_server_c
       break;					// fatal error is set
     MYSQL_QUERY_START(thd->query(), thd->thread_id,
                       (char *) (thd->db ? thd->db : ""),
-                      thd->security_ctx->priv_user,
+                      &thd->security_ctx->priv_user[0],
                       (char *) thd->security_ctx->host_or_ip);
     char *packet_end= thd->query() + thd->query_length();
     /* 'b' stands for 'buffer' parameter', special for 'my_snprintf' */
@@ -1070,7 +1070,7 @@ bool dispatch_command(enum enum_server_c
 
       MYSQL_QUERY_START(beginning_of_next_stmt, thd->thread_id,
                         (char *) (thd->db ? thd->db : ""),
-                        thd->security_ctx->priv_user,
+                        &thd->security_ctx->priv_user[0],
                         (char *) thd->security_ctx->host_or_ip);
 
       thd->set_query_and_id(beginning_of_next_stmt, length,
@@ -5489,7 +5489,7 @@ void mysql_parse(THD *thd, char *rawbuf,
           MYSQL_QUERY_EXEC_START(thd->query(),
                                  thd->thread_id,
                                  (char *) (thd->db ? thd->db : ""),
-                                 thd->security_ctx->priv_user,
+                                 &thd->security_ctx->priv_user[0],
                                  (char *) thd->security_ctx->host_or_ip,
                                  0);
 

=== modified file 'sql/sql_prepare.cc'
--- a/sql/sql_prepare.cc	2010-12-14 10:46:00 +0000
+++ b/sql/sql_prepare.cc	2010-12-08 16:47:21 +0000
@@ -3759,7 +3759,7 @@ bool Prepared_statement::execute(String 
       MYSQL_QUERY_EXEC_START(thd->query(),
                              thd->thread_id,
                              (char *) (thd->db ? thd->db : ""),
-                             thd->security_ctx->priv_user,
+                             &thd->security_ctx->priv_user[0],
                              (char *) thd->security_ctx->host_or_ip,
                              1);
       error= mysql_execute_command(thd);


